src/transformers/modeling_utils.py

```diff
@@ -3560,7 +3347,7 @@ def from_pretrained(
 
             # Check if `_keep_in_fp32_modules` is not None
             use_keep_in_fp32_modules = (cls._keep_in_fp32_modules is not None) and (
-                torch_dtype == torch.float16 or load_in_4bit or load_in_8bit
+                (torch_dtype == torch.float16) or hasattr(hf_quantizer, "use_keep_in_fp32_modules")
```

Comment:
```suggestion
                (torch_dtype != torch.float32) or hasattr(hf_quantizer, "use_keep_in_fp32_modules")
```
but that is for another PR 👿 or maybe this one?

Reply:
Let's do it in another PR ! As discussed offline that feature was used only for models loaded in fp16 to comply with T5 inference. We should do a big refactor of that after this PR gets merged